src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -4156,13 +4156,10 @@ class StubGenerator: public StubCodeGenerator {
     // to minimize the number of memory operations:
     // read the 4 state 4-byte values in pairs, with a single ld,
     // and split them into 2 registers
-    __ mv(t0, mask32);
     __ ld(state0, Address(state));
     __ srli(state1, state0, 32);
-    __ andr(state0, state0, t0);
     __ ld(state2, Address(state, 8));
     __ srli(state3, state2, 32);
-    __ andr(state2, state2, t0);
```

Comment:
Better to add some code comment for this tweak. Or turn the four succeeding `mv` [1] into `addw` with `zr` to make it explicit that the lower half of `state` is needed? 

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/stubGenerator_riscv.cpp#L4170

Reply:
Agree, added some comments.
I don't change `mv` to `addw`, as anyway it will be sign-extened to higher 32 bits, so maybe comment make it more clear.